libbpf-tools/tcppktlat.c

```diff
@@ -236,24 +430,57 @@ int main(int argc, char **argv)
 	}
 
 	if (signal(SIGINT, sig_int) == SIG_ERR) {
-		fprintf(stderr, "can't set signal handler: %s\n", strerror(errno));
+		fprintf(stderr, "can't set signal handler: %s\n",
+			strerror(errno));
 		err = 1;
 		goto cleanup;
 	}
 
-	if (env.timestamp)
-		printf("%-8s ", "TIME(s)");
-	printf("%-7s %-7s %-16s %-*s %-5s %-*s %-5s %-s\n",
-		"PID", "TID", "COMM", column_width, "LADDR", "LPORT", column_width, "RADDR", "RPORT", "MS");
+	if (env.histogram) {
+		printf("Tracing TCP packet latency. Hit Ctrl-C to end.\n");
+	} else {
+		if (env.timestamp)
+			printf("%-8s ", "TIME(s)");
+		printf("%-7s %-7s %-16s %-*s %-5s %-*s %-5s %-s\n", "PID",
+		       "TID", "COMM", column_width, "LADDR", "LPORT",
+		       column_width, "RADDR", "RPORT", "MS");
+	}
 
-	while (!exiting) {
-		err = bpf_buffer__poll(buf, POLL_TIMEOUT_MS);
-		if (err < 0 && err != -EINTR) {
-			fprintf(stderr, "error polling ring/perf buffer: %s\n", strerror(-err));
-			goto cleanup;
+	if (env.histogram) {
+		/* In histogram mode, we don't need to poll events */
+		/* BPF program directly updates the histogram map */
+		if (env.interval < 99999999) {
+			/* Print at specified interval */
+			while (!exiting && env.times > 0) {
+				sleep(env.interval);
+				if (exiting)
+					break;
+				printf("\n");
+				print_hist(obj->maps.hists);
+				if (--env.times == 0)
+					break;
+			}
+		} else {
+			/* Default: wait for Ctrl-C, then print */
+			while (!exiting) {
+				sleep(1);
+			}
+		}
+		/* Print final histogram on exit */
+		printf("\n");
+		print_hist(obj->maps.hists);
```

Comment:
If the tool is run with a 5-second interval and terminated at 13 seconds, the histogram is printed 3 times. However, the time gap between the last two outputs is shorter than the requested 5-second interval. Could you clarify if this behavior is intentional?